cookbooks/aws-parallelcluster-platform/spec/unit/resources/nvidia_driver_spec.rb

```diff
@@ -226,6 +262,110 @@ def self.setup(chef_run, nvidia_driver_version: nil)
               cwd: '/tmp',
               creates: '/usr/bin/nvidia-smi'
             )
+            .with_code(%r{./nvidia.run --silent --dkms --disable-nouveau --no-cc-version-check -m=kernel})
+            .with_code(%r{rm -f /tmp/nvidia.run})
+        end
+      end
+
+      if platform == 'ubuntu'
+        it 'executes initramfs to remove nouveau' do
+          is_expected.to run_execute('initramfs to remove nouveau').with_command('update-initramfs -u')
+        end
+      else
+        it 'does not execute initramfs to remove nouveau' do
+          is_expected.not_to run_execute('initramfs to remove nouveau').with_command('update-initramfs -u')
+        end
+      end
+    end
+
+    context "on #{platform}#{version} when nvidia_driver enabled and node['cluster']['nvidia']['kernel_open'] is true" do
```

Comment:
here, we can have a loop as well, to avoid duplicating all the test code:

```ruby
[%w(false kernel), %w(true kernel-open)].each do |kernel_open, kernel_module|
  context "on #{platform}#{version} when nvidia_driver enabled and node['cluster']['nvidia']['kernel_open'] is #{kernel_open}"
...
allow(res).to receive(:nvidia_kernel_module).and_return(kernel_module)
...
.with_code(%r{CC=/usr/bin/gcc10-gcc ./nvidia.run --silent --dkms --disable-nouveau --no-cc-version-check -m=#{kernel_module}})
...
```
etc